lib/backgroundscanner.php

```diff
@@ -133,6 +93,68 @@ public function run(){
 		$this->tearDownFilesystem();
 	}
 
+	protected function getFilesForScan(){
+		$dirMimeTypeId = \OC::$server->getMimeTypeLoader()->getId('httpd/unix-directory');
+		$qb = \OC::$server->getDatabaseConnection()->getQueryBuilder();
+
+		$sizeLimit = intval($this->appConfig->getAvMaxFileSize());
+		if ( $sizeLimit === -1 ){
+			$sizeLimitExpr = $qb->expr()->neq('fc.size', $qb->expr()->literal('0'));
+		} else {
+			$sizeLimitExpr = $qb->expr()->andX(
+				$qb->expr()->neq('fc.size', $qb->expr()->literal('0')),
+				$qb->expr()->lt('fc.size', $qb->expr()->literal((string) $sizeLimit))
+			);
+		}
+
+		$qb->select(['fc.fileid'])
+			->from('filecache', 'fc')
+			->leftJoin('fc', 'files_antivirus', 'fa', $qb->expr()->eq('fa.fileid', 'fc.fileid'))
+			->innerJoin(
+				'fc',
+				'storages',
+				'ss',
+				$qb->expr()->andX(
+					$qb->expr()->eq('fc.storage', 'ss.numeric_id'),
+					$qb->expr()->orX(
+						$qb->expr()->like('ss.id', $qb->expr()->literal('local::%')),
+						$qb->expr()->like('ss.id', $qb->expr()->literal('home::%'))
```

Comment:
unrelated, but reminded me of a potential issue, raised as https://github.com/owncloud/files_antivirus/issues/137